CppSource/Source/Services/Social/Manager/social_manager.cpp

```diff
@@ -49,13 +57,14 @@ try
     auto cppXboxSocialUsers = group->pImpl->cppSocialUserGroup()->get_users_from_xbox_user_ids(cXboxUserIdContainers);
     *xboxSocialUsersCount = (uint32_t)cppXboxSocialUsers.size();
 
-    socialVars.cXboxSocialUserGroupUsers = std::vector<XSAPI_XBOX_SOCIAL_USER*>(*xboxSocialUsersCount);
+    auto userGroupUsers = get_singleton_for_social()->m_socialVars->cXboxSocialUserGroupUsers;
```

Comment:
>cXboxSocialUserGroupUsers [](start = 68, length = 25)

Does this actually populate the vector in the singleton?  It looks like you are making a copy and populating that one

Reply:
You might need to do auto& usersGroupUsers = ...

---
In reply to: [157635824](https://github.com/Microsoft/xbox-live-unity-plugin/pull/181#discussion_r157635824) [](ancestors = 157635824)